src/library/client.lua

```diff
@@ -33,81 +98,46 @@ function client.get_raw_finale_version(major, minor, build)
 end
 
 --[[
-% supports_smufl_fonts()
+% supports
 
-Returns true if the current client supports SMuFL fonts.
-
-: (boolean)
-]]
-function client.supports_smufl_fonts()
-    return finenv.RawFinaleVersion >= client.get_raw_finale_version(27, 1)
-end
+Checks the client supports a given feature. Returns true if the client
+supports the feature, false otherwise.
 
---[[
-% supports_category_save_with_new_type()
+To assert the client must support a feature, use `client.assert_supports`.
 
-Returns true if the current client supports FCCategory::SaveWithNewType().
+For a list of valid features, see the [`features` table in the codebase](https://github.com/finale-lua/lua-scripts/blob/master/src/library/client.lua#L52).
 
+@ feature (string) The feature the client should support.
 : (boolean)
 ]]
-function client.supports_category_save_with_new_type()
-    return finenv.StringVersion >= "0.58"
+function client.supports(feature)
+    return features[feature].test
```

Comment:
With the latest change, this will no longer throw an error if the feature doesn't exist, is that intentional?

Reply:
Thanks for catching that. I added a check that'll throw an error if the feature is not defined.